### ISSUE TITLE
breaking changes finds wrong errors

### ISSUE THREAD

**Issue by jbreckel** [graphql/graphql-js#783]
~~on the `npm` branch~~ in the 0.9.2 release, `breaking-changes` finds a lot of errors, that shouldn't be errors:

```
{
  "description": "Query.type arg id has changed type from ID! to ID!",
  "type": "ARG_CHANGED_KIND",
},
```

**Comment by wincent** [comment]
Thanks for the report @jbreckel.

Related to https://github.com/graphql/graphql-js/pull/701

What are you doing to repro this error?

**Comment by jbreckel** [comment]
I created a M(n)WE: https://github.com/jbreckel/graphql-breaking-changes-error.
* `yarn run build` to generate a new `schema.json`
* `yarn run test` to test new `schema.js` against `schema.json`

*expected*: should be fine, because I just generated the schema

*actual:* 
```
 FAIL  ./breaking-changes.spec.js
  ● Breaking Changes › there should be no breaking changes

    expect(received).toEqual(expected)

    Expected value to equal:
      Array []
    Received:
      [{"description": "Query.test arg id has changed type from ID! to ID!", "type": "ARG_CHANGED_KIND"}]

    Difference:

    - Expected
    + Received

    -Array []
    +Array [
    +  Object {
    +    "description": "Query.test arg id has changed type from ID! to ID!",
    +    "type": "ARG_CHANGED_KIND",
    +  },
    +]
```

**Comment by jbreckel** [comment]
after some more testing this is what it can be boiled down to:
```
const {
  findBreakingChanges,
  buildClientSchema,
  graphql,
} = require('graphql')

const { introspectionQuery } = require('graphql/utilities')
const { makeExecutableSchema } = require('graphql-tools')

describe('Breaking Changes', () => {
  it('test with graphql-tools', async () => {
    const schemaString = `
    schema {
      query: Query
    }

    type Query {
      test(id: ID!): String
    }
    `

    const Schema = makeExecutableSchema({
      typeDefs: schemaString,
    })

    const { data, errors } = await graphql(Schema, introspectionQuery)
    expect(findBreakingChanges(buildClientSchema(data), Schema)).toEqual([])
  })
})
```

This test works with `graphql@0.9.1` but fails in `graphql@0.9.2` due to the argument of `Query.test`.
It is not only `ID` but every argument is seen as changed and therefore breaking.:

`graphql@0.9.1`:
```
$ yarn upgrade graphql@0.9.1 && yarn test
yarn upgrade v0.21.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved 0 new dependencies.
✨  Done in 1.85s.
yarn test v0.21.3
$ jest
 PASS  ./breaking-changes.spec.js
  Breaking Changes
    ✓ test with graphql-tools (50ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.101s
Ran all test suites.
✨  Done in 1.85s.
```

`graphql@0.9.2`:
```
$ yarn upgrade graphql@0.9.2 && yarn test
yarn upgrade v0.21.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
└─ graphql@0.9.2
✨  Done in 2.37s.
yarn test v0.21.3
$ jest
 FAIL  ./breaking-changes.spec.js
  ● Breaking Changes › test with graphql-tools

    expect(received).toEqual(expected)

    Expected value to equal:
      Array []
    Received:
      [{"description": "Query.test arg id has changed type from ID! to ID!", "type": "ARG_CHANGED_KIND"}]

    Difference:

    - Expected
    + Received

    -Array []
    +Array [
    +  Object {
    +    "description": "Query.test arg id has changed type from ID! to ID!",
    +    "type": "ARG_CHANGED_KIND",
    +  },
    +]

      at it (breaking-changes.spec.js:27:66)
      at process._tickCallback (internal/process/next_tick.js:103:7)

  Breaking Changes
    ✕ test with graphql-tools (54ms)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        1.072s
Ran all test suites.
error Command failed with exit code 1.
```

**Comment by jbreckel** [comment]
most likely not an issue with `graphql/graphql-js` but `apollographql/graphql-tools`:
https://github.com/apollographql/graphql-tools/issues/306

**Comment by joelgriffith** [comment]
Hrm, I'm seeing this as well locally without `graphql-tools`, likely the updates I made to `findBreakingChanges` aren't working as intended...

**Comment by joelgriffith** [comment]
I have a PR in https://github.com/graphql/graphql-js/pull/800 that should address this

**Comment by wincent** [comment]
Just published v0.9.3, which includes that PR. Thanks for the report @jbreckel, and for the fix @joelgriffith.

**Comment by jbreckel** [comment]
thank you 🙂
